base/linalg/cholesky.jl

```diff
@@ -421,19 +421,45 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
 end
 
 function det(C::Cholesky)
-    dd = one(eltype(C))
-    for i in 1:size(C.factors,1); dd *= abs2(C.factors[i,i]) end
+    dd = one(real(eltype(C)))
+    for i in 1:size(C.factors,1)
+        dd *= real(C.factors[i,i])^2
+    end
     dd
 end
 
-det(C::CholeskyPivoted) = C.rank < size(C.factors, 1) ? real(zero(eltype(C))) : prod(abs2.(diag(C.factors)))
-
 function logdet(C::Cholesky)
-    dd = zero(eltype(C))
-    for i in 1:size(C.factors,1); dd += log(C.factors[i,i]) end
+    dd = zero(real(eltype(C)))
+    for i in 1:size(C.factors,1)
+        dd += log(C.factors[i,i])
```

Comment:
Push that change simultaneously with your comment. The line https://github.com/JuliaLang/julia/blob/master/test/linalg/cholesky.jl#L59 should check this and loops over both real and complex matrices.